### ISSUE TITLE
Regarding Prompt for user input

### ISSUE THREAD

**Issue by Lonami** [cli-guidelines/cli-guidelines#38]
Hi, I would like to argue a bit with this point :) 

Unless it's something critical, the CLI should probably just fail. Otherwise, it becomes a problem when scripting, and can even get annoying during normal use.

Later in the post, a `--no-input` flag is suggested. But I would argue interactivity should (generally) be opt-in, so what about a `--interactive` flag instead?

**Comment by aanand** [comment]
I absolutely agree that a prompt for input is an annoyance when a tool is being invoked by a script. But there are heuristics for detecting that case! As we say in the next advice point:

> If `stdin` is not an interactive terminal, skip prompting and just require those flags/args.

As for opt-in versus opt-out, we've planted our flag pretty firmly: [design for humans first](https://clig.dev/#human-first-design). The people who benefit from interactivity outnumber the people who don't - and what's more, they'll likely never discover the `--interactive` flag. The (much smaller) group of people who wrap your command in scripts are much more likely to examine what flags you offer and pass in `--no-input` - and even if they don't, the aforementioned "stdin is a tty" check will catch the majority of cases.

**Comment by MineRobber9000** [comment]
I would argue that the interactive terminal stdin check could be used to tell users about an `--interactive` flag (i.e; if stdin is an interactive terminal and no results are passed in, suggest the user try supplying `--interactive`). Just my two cents, though.

**Comment by 9ary** [comment]
Testing for a tty to change behavior can easily lead to unexpected and undesirable outcomes. It's simply too hard to guess what the caller wants, and is likely to violate expectations.

For example, stdin will still be a tty if the tool is invoked from a shell script that's being run interactively.
I've also removed such a check from a screenshot tool of mine. It would write to a file when stdout is a tty, or dump to stdout otherwise, which I later found out was a bad idea because the tool could be invoked non-interactively, but stdout would still not be redirected, resulting in confusion.

Trying to be smart isn't always a good thing. Failing early with a clear error message is the best way to handle this in my opinion, pressing the up key and adding the missing parameter isn't more inconvenient than answering a prompt, and it makes behavior more predictable while keeping complexity to a minimum.

**Comment by expectocode** [comment]
> pressing the up key and adding the missing parameter isn't more inconvenient than answering a prompt,

To add to this, it also means that if you later invoke the command again (e.g. by searching for it with C-r), it will have the same parameters, making it easier to reproduce previous workflows.

**Comment by aanand** [comment]
I agree that for a single piece of information it's not much more inconvenient to just ask the user to append a flag, but if you need several things from them - e.g. as part of an initial setup process - then it's laborious. Maybe the best approach differs depending on how much information you're asking for.
